src/roswire/common/package.py

```diff
@@ -122,51 +112,39 @@ def build(cls,
         return db_package
 
     @classmethod
-    def _paths_ros1(cls, app_instance: 'AppInstance') -> List[str]:
-        """Parses :code:`ROS_PACKAGE_PATH` for a given shell."""
-        paths: List[str] = []
-        shell = app_instance.shell
-        files = app_instance.files
-        path_str = shell.environ('ROS_PACKAGE_PATH')
-        package_paths: List[str] = path_str.strip().split(':')
-        for path in package_paths:
-            try:
-                all_packages = files.find(path, 'package.xml')
-            except dockerblade.exceptions.DockerBladeException:
-                logger.warning('unable to find directory in ROS_PACKAGE_PATH:'
-                               f' {path}')
-                continue
-            package_dirs = [os.path.dirname(p) for p in all_packages]
-            paths.extend(package_dirs)
-        return paths
-
-    @classmethod
-    def _paths_ros2(cls, app_instance: 'AppInstance') -> List[str]:
-        """Returns a list of paths for all ROS2 packages in an application."""
-        try:
-            shell = app_instance.shell
-            jsn = shell.check_output(_COMMAND_ROS2_PKG_PREFIXES, text=True)
-        except dockerblade.exceptions.CalledProcessError:
-            logger.error('failed to obtain ROS2 package prefixes')
-            raise
-        package_to_prefix: Mapping[str, str] = json.loads(jsn)
-        paths: List[str] = [os.path.join(prefix, f'share/{package}')
-                            for (package, prefix) in package_to_prefix.items()]
-        return paths
-
-    @classmethod
+    @abstractmethod
     def _determine_paths(cls, app_instance: 'AppInstance') -> List[str]:
-        """Parses :code:`ROS_PACKAGE_PATH` for a given shell.
-
+        """
+        Parses the package paths for a given shell.
```

Comment:
Ah. Sorry for the ambiguity. This text should go on the same line as the `"""`. We need the blank line between the method description and the Parameters section header.